src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+    @isTest 
+    static void deleteDefaultAddr() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+       
+        // now let's delete the Addresses
+        system.assertEquals(false, ADDR_Addresses_TDTM.hasRunAddrTrigger);
+        Test.startTest();
+        delete contactsAddrs.addrs;
+        Test.stopTest();
+    
+        // verify that the Contact address fields are cleared
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assertEquals(null, contact.MailingStreet);
+            system.assertEquals(null, contact.MailingCity);
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Delete existing non-default override Addresses.  
+    verify: Contact Mailing address fields not changed.
+    **********************************************************************************************************/            
+    @isTest
+    static void deleteNonDefaultOverrideAddr() {
+        configSettings();
+        
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+        
+        // create additional addresses
+        List<Address__c> addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            addrs[i].Parent_Contact__c = contactsAddrs.contacts[i].Id;
+            addrs[i].Default_Address__c = false;
+            addrs[i].MailingStreet__c = 'override' + i;
+            addrs[i].MailingCity__c = 'override' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        
+        // set the Contacts' address overrides
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            Contact contact = contactsAddrs.contacts[i];
+            contact.Current_Address__c = addrs[i].Id;
+            contact.is_Address_Override__c = true;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;        
+        
+        // now let's delete the override Addresses
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        delete addrs;
+        Test.stopTest();
+    
+        // verify that the Contact address fields went back to the default
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.